pages/options.coffee

```diff
@@ -127,6 +129,19 @@ class ExclusionRulesOption extends Option
   getPassKeys: (element) -> element.querySelector(".passKeys")
   getRemoveButton: (element) -> element.querySelector(".exclusionRemoveButton")
 
+  sortRules: ->
+    rules = @readValueFromElement()
+    for rule in rules
+      key = rule.pattern
+      key = RegExp.$1 if 0 == key.search /^[a-z?*]+:\/\/(.*)/
+      key = RegExp.$1 if 0 == key.search /^www\.(.*)/
+      rule.key = key
+
+    rules.sort (a,b) -> if a.key < b.key then -1 else if b.key < a.key then 1 else 0
+    @element.innerHTML = ""
+    @appendRule rule for rule in rules
+    @onUpdated()
```

Comment:
Having a sort button feels like we're offering too much of an
implementation detail to the user. Chrome's "custom search engines"
preferences page sorts everything by default. What do you think of doing
that?
On Jun 17, 2016 8:28 AM, "Dahan Gong" <notifications@github.com> wrote:

> In pages/options.coffee
> <https://github.com/philc/vimium/pull/2052#discussion_r67528822>:
>
> > @@ -127,6 +129,19 @@ class ExclusionRulesOption extends Option
> >    getPassKeys: (element) -> element.querySelector(".passKeys")
> >    getRemoveButton: (element) -> element.querySelector(".exclusionRemoveButton")
> >
> > +  sortRules: ->
> > +    rules = @readValueFromElement()
> > +    for rule in rules
> > +      key = rule.pattern
> > +      key = RegExp.$1 if 0 == key.search /^[a-z?*]+:\/\/(.*)/
> > +      key = RegExp.$1 if 0 == key.search /^www\.(.*)/
> > +      rule.key = key
> > +
> > +    rules.sort (a,b) -> if a.key < b.key then -1 else if b.key < a.key then 1 else 0
> > +    @element.innerHTML = ""
> > +    @appendRule rule for rule in rules
> > +    @onUpdated()
>
> Maybe we should change the button's text to indicat we have done this work.
>
> My exclusion panel has only 2 rules and both urls are too long for the
> pattern inputs. So I could not see any change after I clicked it.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/philc/vimium/pull/2052/files/5894e17c796dcbb975ff3ee9d71a3c6fc32c16bb#r67528822>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAAhll--1_8amhV-jXajXfwW4W-A1_Pkks5qMr0tgaJpZM4HvkLg>
> .
>